source/Octopus.Tentacle.Tests/Kubernetes/KubernetesDirectoryInformationProviderFixture.cs

```diff
@@ -65,8 +70,52 @@ public void IfDuFailsCompletelyReturnNull()
         {
             var spr = Substitute.For<ISilentProcessRunner>();
             spr.ReturnsForAll(1);
-            var sut = new KubernetesDirectoryInformationProvider(Substitute.For<ISystemLog>(), spr);
+            var memoryCache = new MemoryCache(new MemoryCacheOptions());
+            var sut = new KubernetesDirectoryInformationProvider(Substitute.For<ISystemLog>(), spr, memoryCache);
             sut.GetPathUsedBytes("/octopus").Should().Be(null);
         }
+        
+        [Test]
+        public void ReturnedValueShouldBeCached()
+        {
+            var spr = Substitute.For<ISilentProcessRunner>();
+            spr.ReturnsForAll(1);
+            var memoryCache = new MemoryCache(new MemoryCacheOptions());
+            var sut = new KubernetesDirectoryInformationProvider(Substitute.For<ISystemLog>(), spr, memoryCache);
+            sut.GetPathUsedBytes("/octopus").Should().Be(null);
+            
+            const ulong usedSize = 500 * Megabyte;
+            spr.When(x => x.ExecuteCommand("du", "-s -B 1 /octopus", "/", Arg.Any<Action<string>>(), Arg.Any<Action<string>>()))
+                .Do(x =>
+                {
+                    x.ArgAt<Action<string>>(3).Invoke($"500\t/octopus");
+                    x.ArgAt<Action<string>>(3).Invoke($"{usedSize}\t/octopus");
+                });
+            
+            sut.GetPathUsedBytes("/octopus").Should().Be(null);
+        }
+        
+        [Test]
+        public void DuCacheExpiresAfter15Seconds()
+        {
+            var spr = Substitute.For<ISilentProcessRunner>();
+            spr.ReturnsForAll(1);
+            var memoryCache = new MemoryCache(new MemoryCacheOptions());
```

Comment:
`MemoryCacheOptions` has a `Clock` property where you can override the default clock for testing. This way we won't need to call `Thread.Sleep()`